Templates/cmake/SYCL/target_link_library/CMakeLists.txt

```diff
@@ -0,0 +1,30 @@
+#==============================================================
+# Copyright © Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+#=============================================================
+
+if (CMAKE_HOST_WIN32)
+    # need CMake 3.25.0+ for IntelLLVM support of target link properties on Windows
+    cmake_minimum_required(VERSION 3.25)
+else()
+    # CMake 3.20.5 is the minimum recommended for IntelLLVM on Linux
+    cmake_minimum_required(VERSION 3.20.5)
+endif()
+
+# The Modern CMake way to use separate package is to set up CMake to use the
+# package with `find_package`, then to add it to the project with
+# `target_link_libraries`.  This project demonstrates how to build a SYCL
+# target with Intel oneAPI compilers.
+#
+# While consistent with Modern CMake, this method does not work for other
+# SYCL implementations.  Also, every source file in the specified target will
+# be compiled for offload.  Compiling with offload enabled will not affect
+# correctness but does lead to substantially longer compile times.
+project(sycl-target-link-libraries LANGUAGES CXX)
+
+find_package(IntelSYCL REQUIRED)
+
+# Add SYCL properties to the whole target, the modern CMake way
+add_executable(simple-tll simple.cpp)
+target_link_libraries(simple-tll PRIVATE IntelSYCL::SYCL_CXX)
```

Comment:
What does this `IntelSYCL::SYCL_CXX` option do?

Reply:
That interface imports Intel specific sycl options.
 The ` target_link_libraries(..IntelSYCL::SYCL_CXX) `   behaves same way as   `add_sycl_to_target()` but instead of applying the settings per file,  it applies to the whoe project.  While we prefer the `add_sycl_to_target()` way of applying intel sycl specific options, we do have support for ` target_link_libraries(..IntelSYCL::SYCL_CXX) `  as some other Sycl vendors provide it.

Code in `bin/../IntelSYCL/IntelSYCLConfig.cmake` .

Reply:
would it make sense to merge the two samples and just have the `add_sycl_to_target()` one have a comment saying: alternatively you can use `target_link_libraries(..IntelSYCL::SYCL_CXX)` to specify that... ?